save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/TopBarUrl.kt

```diff
@@ -0,0 +1,158 @@
+    }
+
+    /**
+     * The function classification url address
+     *
+     * @param href
+     */
+    private fun findExclude(href: String) {
+        circumstance = if (href.contains(SituationUrlClassification.ORGANIZATION.regex)) {
+            SituationUrlClassification.ORGANIZATION
+        } else if (href.contains(SituationUrlClassification.PROJECT.regex)) {
+            SituationUrlClassification.PROJECT
+        } else if (href.contains(SituationUrlClassification.ARCHIVE.regex)) {
+            SituationUrlClassification.ARCHIVE
+        } else if (href.contains(SituationUrlClassification.DETAILS.regex)) {
+            SituationUrlClassification.DETAILS
+        } else if (href.contains(SituationUrlClassification.EXECUTION.regex)) {
+            SituationUrlClassification.EXECUTION
+        } else {
+            SituationUrlClassification.KEYWORD_PROCESS
+        }
+    }
+
+    /**
+     * Function check exclude and generate currentPath after the buttons creating
+     *
+     * @param pathPart
+     */
+    private fun fixCurrentPathAfter(pathPart: String) {
+        currentPath = when (circumstance) {
+            SituationUrlClassification.PROJECT, SituationUrlClassification.ORGANIZATION, SituationUrlClassification.ARCHIVE -> "#"
+            SituationUrlClassification.DETAILS, SituationUrlClassification.EXECUTION -> if (pathPart == "execution") mergeUrls(pathPart) else currentPath
+            else -> currentPath
+        }
+    }
+
+    /**
+     * The function changes the exclude after the button is created
+     *
+     * @param isNumber
+     */
+    private fun fixExcludeAfter(isNumber: Boolean) {
+        circumstance = when (circumstance) {
+            SituationUrlClassification.PROJECT, SituationUrlClassification.ORGANIZATION, SituationUrlClassification.ARCHIVE -> SituationUrlClassification.KEYWORD_PROCESS
+            SituationUrlClassification.DETAILS -> if (isNumber) setProcessLastSegments(1) else SituationUrlClassification.DETAILS
+            else -> circumstance
+        }
+    }
+
+    private fun mergeUrls(secondPath: String) = "$currentPath/$secondPath"
+
+    private fun setProcessLastSegments(number: Int): SituationUrlClassification {
+        processLastSegments = number
+        return SituationUrlClassification.KEYWORD_PROCESS_LAST_SEGMENTS
+    }
+
+    /**
+     * This Enum class classifies work with the url address segment
+     * @property regex
+     */
+    enum class SituationUrlClassification(val regex: Regex = Regex("")) {
+        /**
+         * Situation with the processing of the "archive" in the url address - need for tabs in AwesomeBenchmarksView
+         */
+        ARCHIVE(BenchmarkCategoryEnum.regexForUrlClassification),
+
+        /**
+         * Situation with the processing of the "details" in the url address - need for deleted multi-segment urls, starting with the word "details"
+         */
+        DETAILS(Regex("/[^/]+/[^/]+/history/execution/[1234567890]+/details")),
```

Comment:
btw, digitals can be matched by `[0-9]+` or `[\d]+`

Reply:
Done